share/models/core.py

```diff
@@ -75,6 +81,37 @@ def create_robot_user(self, username, robot, long_title='', home_page=''):
         return user
 
 
+class FaviconImage(models.Model):
+    user = models.OneToOneField('ShareUser', on_delete=DATABASE_CASCADE, primary_key=True)
+    image = models.BinaryField()
+
+
+@deconstructible
+class FaviconStorage(Storage):
+    def _open(self, name, mode='rb'):
+        assert mode == 'rb'
+        favicon = FaviconImage.objects.get(user__username=name)
+        return ContentFile(favicon.image)
+
+    def _save(self, name, content):
+        user = ShareUser.objects.get(username=name)
```

Comment:
I'd tried `user__username=name` but it complained that `user_id` was missing. Apparently it's not smart enough for that.